jetty-server/src/main/java/org/eclipse/jetty/server/ServletAttributes.java

```diff
@@ -0,0 +1,64 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.util.Set;
+
+import org.eclipse.jetty.util.Attributes;
+import org.eclipse.jetty.util.AttributesMap;
+
+public class ServletAttributes implements Attributes
+{
+    private final Attributes _attributes;
+
+    public ServletAttributes()
+    {
+        _attributes = new AsyncAttributes(new AttributesMap());
+    }
```

Comment:
I don't like all the dereferences here.   We end up with `ServletAttributes->AsyncAttributes->AttributesMap`, so 3 object and 3 dereferences.
I think it would be much better to do (also note the name change):
```suggestion
public class AsyncAttributesMap extends AttributesMap
{
    public AsyncAttributesMap()
    {
    }
```
The override the methods as follows:
```java
    @Override
￼    public Object getAttribute(String name)
￼    {
        // do the AsyncAttribute handling here
        // else
        return super.getAttribute(name);
    }
```

Then you don't need AsyncAttributes class.